### ISSUE TITLE
Improve accessible `Name` of the Navigation Offcanvas List View

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47022]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1374652632 the accessible `Name` of the List View should be `Navigation Menu: %%NAV MENU NAME%%` (please consider when we don't have a menu name or it's an auto menu). 

Currently it's something about "Application" which is confusing.

**Comment by getdave** [comment]
I think I accidentally duplicated this Issue here https://github.com/WordPress/gutenberg/issues/47010. I'l close that one.
